uncaged-cli/main.go

```diff
@@ -42,29 +48,66 @@ type UncagedCLI struct {
 	bookDir      string
 	metadataFile string
 	drivinfoFile string
-	metadata     []map[string]interface{}
+	metadata     cliMeta
 	deviceInfo   uc.DeviceInfo
 }
 
+type cliMeta struct {
```

Comment:
It might be a good idea to use a mutex here, just in case this is ever accessed concurrently.

Reply:
Hmm. Actually, looking at that code again, might be better to generate a one-time use iterator instead. That way multiple goroutines could iterate over the metadata independently without issues. Thanks for pointing that out.

Just as a side-note, the code in the cli path isn't really production quality. It's what I use to quickly test new features/changes in the library.